test/metabase/lib/filter_test.cljc

```diff
@@ -181,3 +181,87 @@
                                 [:= venues-category-id-metadata 242 categories-id-metadata]
                                 [:contains venues-name-metadata "part"]]])
                   (dissoc :lib/metadata)))))))
+
+(deftest ^:parallel replace-filter-test
+  (let [q1                          (lib/query-for-table-name meta/metadata-provider "CATEGORIES")
+        venues-name-metadata        (lib.metadata/field q1 nil "VENUES" "NAME")
+        between-uuid (str (random-uuid))]
+    (let [simple-filtered-query
+        {:lib/type :mbql/query,
+         :database (meta/id)
+         :type :pipeline
+         :stages [{:lib/type :mbql.stage/mbql
+                   :source-table (meta/id :categories)
+                   :lib/options {:lib/uuid (str (random-uuid))},
+                   :filter [:between
+                            {:lib/uuid between-uuid}
+                            [:field
+                             {:base-type :type/Integer, :lib/uuid (str (random-uuid))}
+                             (meta/id :venues :category-id)]
+                            42
+                            100]}]}
+          result-query
+          (assoc-in simple-filtered-query
+                               [:stages 0 :filter]
+                               [:starts-with
+                                {:lib/uuid string?}
+                                [:field {:base-type :type/Text, :lib/uuid string?} (meta/id :venues :name)]
+                                "part"])]
```

Comment:
It would be good to dogfood stuff in `metabase.lib.core` to build queries instead of writing out the MBQL by hand